File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        dic=defaultdict(int)
        ans=""
        for i in s:
            dic[i]+=1
        d=sorted(dic.items(),key=lambda x:-x[1])
        for i,v in d:
            ans+=i*v
        return ans
        

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        map<char,int>mp;
        for(int i=0;i<s.size();i++){
            mp[s[i]]++;
        }
        priority_queue<pair<int,char>>pq;
        for(auto i:mp){
            pq.push({i.second,i.first});
        }
        string ans="";
        while(!pq.empty()){
            int size=pq.top().first;
            for(int j=0;j<size;j++){
                ans+=pq.top().second;
            }
            pq.pop();
        }
        return ans;
    }
};